ci/benchmark_bencher.yml

```diff
@@ -0,0 +1,49 @@
+include:
+  - local: 'ci/base.yml'
+
+.variables:
+  PR_ID: $(echo "${CI_COMMIT_BRANCH}" | grep -o 'pr[0-9]*' | grep -o '[0-9]*')
+  FEATURE_BRANCH: $(curl -s https://api.github.com/repos/C2SM/icon4py/pulls/$PR_ID | jq -r '.head.ref')
+  GITHUB_ACTIONS: true # needed for GitHub PR comments - along with --github-actions -
+  GITHUB_EVENT_NAME: pull_request # needed for GitHub PR comments - along with --github-actions -
+  GITHUB_STEP_SUMMARY: $CI_PROJECT_DIR/step_summary.log # needed for GitHub PR comments - along with --github-actions -
+  GITHUB_SHA: $CI_COMMIT_SHA # needed for GitHub PR comments - along with --github-actions -
+  GITHUB_EVENT_PATH: $CI_PROJECT_DIR/event.json # needed for GitHub PR comments - along with --github-actions -
+
+.bencher_github_actions_setup:
+  script:
+    - |
+      echo "{\"pull_request\": {\"head\": {\"repo\": {\"full_name\": \"C2SM/icon4py\"}}}, \"repository\": {\"full_name\": \"C2SM/icon4py\"}, \"number\": $PR_ID}" > $CI_PROJECT_DIR/event.json
+
+.install_bencher:
+  script:
+    - rustup default stable
+    - cargo install --git https://github.com/finkandreas/bencher --branch main --locked --force bencher_cli --root /root/.cargo # Specific version of bencher that supports --github-actions
```

Comment:
I thought I have added another question in my review but maybe didn't click the button... Anyway, the question is if there is a roadmap to integrate the changes from Andreas' branch in upstream bencher, because otherwise someone needs to keep syncing the fork with upstream, and it may break at any point, so it's not an ideal situation.

Reply:
Andreas is waiting for a response from Bencher developers (opened an issue) about how to proceed. Indeed, the plan is to open a PR and merge his changes. Although, the changes are only for the GitHub commenting, so they are not very critical on how we use Bencher. Worst case scenario is to use the actual Bencher without GitHub  commenting.

Reply:
Ok, thanks for the explanation. I would expand the comment to explain this in the file, and maybe add a TODO to check again soon when we should move back to upstream bencher